src/Migrations.php

```
@@ -0,0 +1,280 @@
+     * Available options are :
+     * - `connection` The datasource connection to use
+     * - `source` The folder where migrations are in
+     * - `plugin` The plugin containing the migrations
+     */
+    public function __construct(array $default = [])
+    {
+        $this->output = new NullOutput();
+
+        if (!empty($default)) {
+            $this->default = $default;
+        }
+    }
+
+    /**
+     * Returns the status of each migrations based on the options passed
+     *
+     * @param array $options Options to pass to the command
+     * Available options are :
+     *
+     * - `format` Format to output the response. Can be 'json'
+     * - `connection` The datasource connection to use
+     * - `source` The folder where migrations are in
+     * - `plugin` The plugin containing the migrations
+     *
+     * @return array The migrations list and their statuses
+     */
+    public function status($options = [])
+    {
+        $input = $this->getInput('Status', [], $options);
+        $params = ['default', $input->getOption('format')];
+
+        return $this->run('printStatus', $params, $input);
+    }
+
+    /**
+     * Migrates available migrations
+     *
+     * @param array $options Options to pass to the command
+     * Available options are :
+     *
+     * - `target` The version number to migrate to. If not provided, will migrate
+     * everything it can
+     * - `connection` The datasource connection to use
+     * - `source` The folder where migrations are in
+     * - `plugin` The plugin containing the migrations
+     *
+     * @return bool Success
+     */
+    public function migrate($options = [])
+    {
+        $input = $this->getInput('Migrate', [], $options);
+        $params = ['default', $input->getOption('target')];
+
+        try {
+            $this->run('migrate', $params, $input);
+            return true;
+        } catch (\Exception $e) {
```

Would it be too bad to just let the exception bubble up? I think it is better than jus returning false